pkg/controller/priorityqueue/priorityqueue.go

```diff
@@ -326,36 +331,64 @@ func (w *priorityqueue[T]) Len() int {
 	return result
 }
 
+func (w *priorityqueue[T]) logState() {
+	t := time.Tick(10 * time.Second)
+	for {
+		select {
+		case <-w.done:
+			return
+		case <-t:
+		}
+
+		// Log level may change at runtime, so keep the
+		// loop going even if a given level is currently
+		// not enabled.
+		if !w.log.V(1).Enabled() {
```

Comment:
I know its a bit more common in the ecosystem to use `V(4)` for `debug`, but I think many of our users use `zap` + `zapr` as the underlying implementation for the `logr.Logger`, given that we include it and do that in our examples. The problem is that `zap` has named log levels. These do end up mapping to a `int8` but I personally only learned this recently and I've been using `zap` for years. The named `debug` level maps to `-1` and `zapr`  then switches the sign, resulting in `V(1)` in the `logr.Logger`.

Unless we have a strong reason why we need something higher than `V(1)`, i'd like to keep it this way so that this log shows up if someone uses a `zap` logger on `debug` level.

Reply:
The recommended production log level with klog is 2

Reply:
I've updated it to `V(5)` for now so this doesn't show up by default and opened an issue for what seems to be a mismatch for the log level mapping: https://github.com/kubernetes-sigs/controller-runtime/issues/3081